### ISSUE TITLE
Improve factor-selection-via-machine-learning with pooled estimation

### ISSUE THREAD

**Issue by christophscheuch** [tidy-finance/website#268]
## Motivation

The chapter teaches the `tidymodels` + `glmnet` workflow really well, and I'd like to keep that pedagogy fully intact. My concern is with the empirical application: it runs a **separate Lasso per industry** and then asks which variables survive in each one. The chapter's own punchline is that "Lasso does not provide a factor that consistently provides pricing abilities across industries." I think that lack of consistency is largely an artifact of the setup rather than an economic finding, and that a small reframing makes the chapter both more correct and a stronger payoff for the factor-zoo motivation.

## Why the current application doesn't quite work

In asset pricing, whether a factor is *priced* (i.e., whether it enters the SDF) is a property of the market, not of any single test asset. Estimating an independent factor set for each of the 10 industries implicitly assumes 10 different pricing kernels, which sits awkwardly next to the CAPM / Fama-French / Fama-MacBeth machinery developed earlier in the book. Concretely:

- We end up with 10 separate "factor zoos" rather than one coherent answer to the chapter's headline question.
- The heatmap of heterogeneous per-industry selection reads like a result, but it's mostly a consequence of misspecification plus per-industry sampling noise.
- The FF3 + macro-predictor design also balloons into 126 interaction columns, which adds incidental complexity without much economic payoff.

## Proposed application: one pooled SDF, industry-specific loadings

Estimate a **single** SDF on the pooled panel of all industries, and let industries differ only in their **loadings**:

$$r_{i,t}^e = \alpha_i + \sum_k \beta_{i,k}\, f_{k,t} + \varepsilon_{i,t}$$

- $\alpha_i$: industry dummies (one intercept per industry)
- $\beta_{i,k}$: industry by factor interactions (the loadings that vary by industry)
- a single Elastic Net penalty over **all** interactions, so a useless factor gets pushed to zero across all industries at once

The clue is that a shared penalty over all industry interactions answers the right question: which factors does the *whole panel* need? What survives is the common SDF; the heterogeneity lives in the loadings, not in different factor sets per industry.

## Concrete changes

- **Data:** swap FF3 + Goyal-Welch macro predictors for **FF5 + q-factors** as a cleaner factor horse-race on the same 10 industry test assets. This is secondary to the SDF reframing, and macro predictors could stay as an optional extension if desired.
- **Split:** `initial_time_split()` on the whole panel.
- **Recipe:** build the SDF specification directly in the recipe.

```r
rec <- recipe(ret_excess ~ ., data = training(split)) |>
  step_rm(date) |>
  step_dummy(industry, one_hot = TRUE) |>
  step_interact(~ starts_with("industry_"):starts_with("factor_")) |>
  step_rm(starts_with("factor_"))
```

- **Model:** a single Elastic Net (`penalty = tune()`, `mixture = tune()`), one `tune_grid()` over the pooled time-series folds, then `select_best()`, `finalize_workflow()`, final `fit()`. This replaces the parallelized per-industry loop entirely.
- **Output:** a heatmap of the **SDF loadings** per industry, grouped by market / FF5 / q. Instead of black-and-white "selected vs. not," show *signed* loadings with a diverging fill. That surfaces economically sensible patterns, e.g. the market factor priced everywhere with utilities defensive (beta around 0.6) and durables/tech cyclical (around 1.1), and lets FF5 and q be compared side by side.

Key figure:

<img width="1920" height="960" alt="Image" src="https://github.com/user-attachments/assets/e6fb671f-101c-4d32-a60f-4b8ae3c59872" />

## What stays the same

The chapter's structure and learning goals are unchanged: the `tidymodels` workflow, recipes / `prep()` / `bake()`, the `parsnip` + `glmnet` engine, time-series cross-validation, and hyperparameter tuning. The only thing that changes is the empirical framing.

**Comment by voigtstefan** [comment]
I like the change. Suggest to implement on a branch pointing to prep/unified-chapters directly to avoid overwriting the changes once we relaunch the website

**Comment by voigtstefan** [comment]
"so a useless factor gets pushed to zero across all industries at once"

Thinking about it a bit more, this statement is actually not correct in the current specification: roe may be irrelevant for manufacturing but matters for durable goods (as of your heatmap). That is where Kozak-Nagel propose a two-step estimator which essentially pushes factor risk premia to zero (lasso) but I am not entirely convinced we want this separate story in our chapter on ML. So the correct statement is more: Instead of doing a similar regression with potentially industry-specific shrinkage intensity lambda, we select one fitting regression simultaneously.

**Comment by christophscheuch** [comment]
Maybe it's worth pivoting this chapter completely because it is really dense and the industry angle is tricky. Maybe we switch to stock-level data and give us some more space to explain the real issues:

- Estimating the Stochastic Discount Factor: Build the SDF using the FF5 factors, priced out-of-sample against a broad CRSP cross-section. Show that the sample-based estimate performs poorly out of sample and degrades as q factors are added, exposing estimation error as the central problem the next two chapters address.

- Shrinking the SDF with Lasso and Ridge: Introduce L1 and L2 penalties as two answers to that estimation error, and frame the contrast economically: Lasso assumes a sparse SDF (few true factors), Ridge a dense one (many factors each mattering a little). Trace coefficient paths and OOS Sharpe against the penalty for both, noting that picking the peak by eye is peeking.

- Factor Selection with Machine Learning: Generalize to the elastic net, whose grouping effect handles the near-duplicate factors (RMW/ROE, CMA/investment) that destabilize Lasso. With two hyperparameters to set, motivate cross-validation as the honest alternative to eyeballing the test set, and tune lambda and rho properly to deliver the OOS payoff.

Or is this computationally too demanding?
